ads/alp/handler.js

```diff
@@ -181,3 +199,51 @@ export function warmupDynamic(e) {
 function getHeadOrFallback(doc) {
   return doc.head || doc.documentElement;
 }
+
+/**
+ * Returns info about an ancestor that can perform A2A navigations
```

Comment:
The current implementation implies the ancestor has to be top ancestor. Is the doc here correct? Or later it can be any ancestor?